src/edge_collections.rs

```diff
@@ -0,0 +1,52 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2022
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use numpy::IntoPyArray;
+use pyo3::prelude::*;
+use pyo3::Python;
+
+/// A simple container that contains a vector representing edges in the
+/// coupling map that are found to be optimal by the swap mapper.
+#[pyclass(module = "stoachstic_swap")]
+#[pyo3(text_signature = "(/")]
+#[derive(Clone, Debug)]
+pub struct EdgeCollection {
+    pub edges: Vec<usize>,
+}
+
+impl Default for EdgeCollection {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[pymethods]
+impl EdgeCollection {
+    #[new]
+    pub fn new() -> Self {
+        EdgeCollection { edges: Vec::new() }
+    }
+
+    /// Add two edges, in order, to the collection.
+    ///
+    /// Args:
+    ///     edge_start (int): The beginning edge.
+    ///     edge_end (int): The end of the edge.
```

Comment:
Yeah it does, the pyo3 macros (like `#[pyfunction]`, `#[pyclass]`, and `#[pymethods]`) will take the rust docstring which is 3 slashes (instead of 2 for a comment) and just set it to `__doc__` attribute for the object it exposes over the python C api ffi they generate. So it should just work fine when sphinx or anything else parses the docs for the compiled module. As for this specific docstring I actually just copied it from the cython.

Reply:
Agree that we're unlikely to publish this publicly.  In that case, will IDE tooling generally pick up this style of docs?  It might be worth writing in rustdoc style to gain those benefits.

Reply:
I don't actually know. I think probably not because there isn't actually a standardized format for arguments in rustdoc (see https://github.com/rust-lang/rust/issues/57525 for the tracking issue on that feature). Most projects just do internal links and lists to have standardized looking thing. My editor just shows me the function signature with the types so I can tab complete arguments when calling something.